testing/xfpga/test_properties_c.cpp

```diff
@@ -74,32 +73,98 @@ class properties_p1 : public ::testing::TestWithParam<std::string> {
   std::array<fpga_token, 2> tokens_;
   fpga_properties filter_;
   fpga_properties props_;
-  fpga_handle handle_;
   fpga_handle accel_;
   uint32_t num_matches_;
   test_platform platform_;
   test_device invalid_device_;
   test_system* system_;
 };
 
-TEST_P(properties_p1, from_handle) {
+/**
+ * @test    from_handle
+ * @brief   Tests: xfpga_fpgaGetPropertiesFromHandle
+ * @details Given a null fpga_properties* object<br>
+ *          When I call xfpga_fpgaGetPropertiesFromHandle with an valid handle,
+ *          expected result is FPGA_OK.<br>
+ */
+TEST_P(properties_c_p, from_handle) {
   EXPECT_EQ(xfpga_fpgaGetPropertiesFromHandle(accel_, &props_), FPGA_OK);
 }
 
+/**
+ * @test       vendor_id
+ *
+ * @brief      When querying the vendor ID of an AFU,
+ * 	       0x8086 is returned.
+ */
+TEST_P(properties_c_p, vendor_id) {
+#ifndef BUILD_ASE
+  auto device = platform_.devices[0];
+  uint16_t x = 0;
+  ASSERT_EQ(xfpga_fpgaGetPropertiesFromHandle(accel_, &props_), FPGA_OK);
+  ASSERT_EQ(FPGA_OK, fpgaPropertiesGetVendorID(props_, &x));
+  EXPECT_EQ(x, device.vendor_id);
+#endif
+}
+
+/**
+ * @test       device_id
+ *
+ * @brief      When querying the device ID of an AFU,
+ * 	       0xbcc0 is returned.
+ */
+TEST_P(properties_c_p, device_id) {
+#ifndef BUILD_ASE
+  auto device = platform_.devices[0];
+  uint16_t x = 0;
+  ASSERT_EQ(xfpga_fpgaGetPropertiesFromHandle(accel_, &props_), FPGA_OK);
+  ASSERT_EQ(FPGA_OK, fpgaPropertiesGetDeviceID(props_, &x));
+  EXPECT_EQ(static_cast<uint32_t>(x), device.device_id);
+#endif
+}
+
+/**
+ * @test       valid_get
+ *
+ * @brief      When fpgaGetPropertiesFromHandle is called
+ *             with a valid handle,
+ *             the function returns FPGA_OK, and
+ *             the returned properties are valid for that handle.
+ */
+TEST_P(properties_c_p, valid_gets) {
+  fpga_objtype objtype;
+
+  ASSERT_EQ(FPGA_OK, xfpga_fpgaGetPropertiesFromHandle(accel_, &props_));
+  ASSERT_EQ(FPGA_OK, fpgaPropertiesGetObjectType(props_, &objtype));
+  EXPECT_EQ(FPGA_ACCELERATOR, objtype);
+}
```

Comment:
This test is missing the case for FME. And probably should explicitly set the object type to something that is not expected, prior to getting the object type in properties.